python/cudf/cudf/dataframe/series.py

```diff
@@ -464,62 +461,56 @@ def tail(self, n=5):
-            Maximum number of rows to show.
-            If it is None, all rows are shown.
+        Uses Pandas formatting internals to produce output identical to Pandas.
+        Use the Pandas formatting settings directly in Pandas to control cuDF
+        output.
         """
-        if nrows is NOTSET:
-            nrows = settings.formatting.get(nrows)
+        return self.__repr__()
 
-        str_dtype = self.dtype
+    def __str__(self):
+        return self.to_string()
 
-        if len(self) == 0:
-            return "<empty Series of dtype={}>".format(str_dtype)
+    def __repr__(self):
+        mr = pd.options.display.max_rows
+        if len(self) > mr:
+            top = self.head(int(mr))
+            bottom = self.tail(int(mr))
+            from cudf import concat
 
-        if nrows is None:
-            nrows = len(self)
+            preprocess = concat([top, bottom])
         else:
-            nrows = min(nrows, len(self))  # cap row count
-
-        more_rows = len(self) - nrows
-
-        # Prepare cells
-        cols = OrderedDict([("", self.values_to_string(nrows=nrows))])
-        dtypes = OrderedDict([("", self.dtype)])
-        # Format into a table
-        output = formatting.format(
-            index=self.index,
-            cols=cols,
-            dtypes=dtypes,
-            more_rows=more_rows,
-            series_spacing=True,
-        )
-        if self.name is None:
-            output += "\ndtype: {}".format(str_dtype)
+            preprocess = self
+        if is_categorical_dtype(preprocess.dtype):
+            return preprocess.to_pandas().__repr__()
+        if preprocess.has_null_mask and not preprocess.dtype == "O":
+            output = (
+                preprocess.astype("O").fillna("null").to_pandas().__repr__()
+            )
         else:
-            output += "\nName: {}, dtype: {}".format(self.name, str_dtype)
-
-        if is_categorical_dtype(self.dtype):
-            cat = self.cat
-            str_dtype = cat.categories.dtype.name
-            categories = cat.categories.to_array()
-            separator = " < " if cat.ordered else ", "
-            desc = "({}, {})".format(len(categories), str_dtype)
-            categories = "[{}]".format(separator.join(categories))
-            output += "\nCategories {}: {}".format(desc, categories)
-
-        return output
-
-    def __str__(self):
-        return self.to_string(nrows=10)
-
-    def __repr__(self):
-        return "<cudf.Series nrows={} >".format(len(self))
+            output = preprocess.to_pandas().__repr__()
+        lines = output.split("\n")
+        if len(lines) > 1:
+            if lines[-1].startswith("N"):
+                lines = lines[:-1]
+                lines.append("Name: %s" % str(self.name))
+                if len(self) > len(preprocess):
+                    lines[-1] = lines[-1] + ", Length: %d" % len(self)
+                lines[-1] = lines[-1] + ", "
+            elif lines[-1].startswith("L"):
```

Comment:
This too

Reply:
Check against sizes here instead of just looking at string? I'm not sure it is worth the trouble. Why don't we solve that problem when it is a problem, instead of prematurely optimizing it?

Reply:
What if the Series has both a name and a Length in the string output?

Reply:
```suggestion
            elif lines[-1].startswith("Length: "):
```

Reply:
👍